src/main/java/fr/insalyon/creatis/agent/workflow/database/JobDatabase.java

```diff
@@ -22,27 +25,36 @@ public JobDatabase(String workflowsPath, String workflowId) {
         this.connect();
     }
 
-    public List<String> getJobIds() {
+    public Map<String,String> getJobsToKill() {
         try {
-            PreparedStatement ps = this.connection.prepareStatement("SELECT id FROM Jobs WHERE status='SUCCESSFULLY_SUBMITTED' OR status='QUEUED' OR status='RUNNING'");
+            PreparedStatement ps = this.connection.prepareStatement("SELECT id, status FROM Jobs WHERE status='SUCCESSFULLY_SUBMITTED' OR status='QUEUED' OR status='RUNNING'");
             ResultSet rs = ps.executeQuery();
-            ArrayList jobIds = new ArrayList();
+            Map<String,String> jobs = new HashMap<>();
 
             while(rs.next()) {
-                jobIds.add(rs.getString("id"));
+                jobs.put(rs.getString("id"),rs.getString("status"));
             }
 
-            return jobIds;
+            return jobs;
         } catch (SQLException e) {
             logger.error("Error getting job ids", e);
             return null;
         }
     }
 
-    public void updateStatus(String jobId) {
+
+    public void updateStatusToCancelled(String jobId) {
+        updateStatus(jobId, "CANCELLED");
+    }
+
+    public void updateStatusToDeleted(String jobId) {
+        updateStatus(jobId, "DELETED");
+    }
+
+    public void updateStatus(String jobId, String status) {
         try {
             PreparedStatement ps = this.connection.prepareStatement("UPDATE Jobs SET status = ? WHERE id = ?");
-            ps.setString(1, "CANCELLED");
+            ps.setString(1, status);
```

Comment:
Have you checked the statuses supported by dirac (https://github.com/DIRACGrid/DIRAC/blob/c081ff19ab706fd9fef5d9c923b52d0e6aaa4e86/src/DIRAC/WorkloadManagementSystem/Client/JobStatus.py#L96)? What happens if there's a status unknown to the agent?